kb/2023-08-23/using_rook_ceph_storage.md

```diff
@@ -0,0 +1,86 @@
+
+Rook automates deployment and management of Ceph to provide self-managing, self-scaling, and self-healing storage services. The Rook operator does this by building on Kubernetes resources to deploy, configure, provision, scale, upgrade, and monitor Ceph.
+
+This article explains how to install, configure, and use Rook Ceph as a data disk for a VM in Harvester.
+
+## Install Harvester Cluster
+
+Harvester's OS has an immutable design, which means most files in the  OS revert to their pre-configured state after a reboot.
+
+So we need to add some persistent paths used by Rook Ceph to `os.persistentStatePaths` in the [Harvester configuration](https://docs.harvesterhci.io/dev/install/harvester-configuration#ospersistent_state_paths).
+
+```yaml
+os:
+  persistentStatePaths:
+    - /var/lib/rook
+    - /var/lib/ceph
+  modules:
+    - rbd
+    - nbd
+```
+
+After the cluster is installed, refer to [How can I access the kubeconfig file of the Harvester cluster?](https://docs.harvesterhci.io/v1.1/faq#how-can-i-access-the-kubeconfig-file-of-the-harvester-cluster) to get the kubeconfig of the Harvester cluster.
+
+## Install Rook Ceph to Harvester
+
+Next, install Rook Ceph to the Harvester cluster by referring to the [Rook Ceph Quickstart](https://rook.io/docs/rook/v1.12/Getting-Started/quickstart/).
+
+1. Install Rook operator
+```bash
+curl -fsSLo rook.tar.gz https://github.com/rook/rook/archive/refs/tags/v1.12.2.tar.gz \
+  && tar -zxf rook.tar.gz && cd rook-1.12.2/deploy/examples
+# apply configurations ref: https://rook.github.io/docs/rook/v1.12/Getting-Started/example-configurations/
+kubectl apply -f crds.yaml -f common.yaml -f operator.yaml
+kubectl -n rook-ceph wait --for=condition=Available deploy rook-ceph-operator --timeout=10m
+```
+
+2. Create Rook Ceph Cluster
+```bash
+# apply configurations ref: https://rook.github.io/docs/rook/v1.12/Getting-Started/example-configurations/#cluster-crd
+kubectl apply -f cluster.yaml
+```
+
+3. Create StorageClass `rook-ceph-block` and VolumeSnapshotClass `csi-rbdplugin-snapclass`
+```bash
+kubectl apply -f ./csi/rbd/storageclass.yaml -f ./csi/rbd/snapshotclass.yaml
+```
+
+## Configure Harvester Cluster
+
+We must configure some necessary information through the Harvester [csi-driver-config](https://docs.harvesterhci.io/dev/advanced/settings#csi-driver-config) setting before using Harvester **Backup/Snapshot** related features.
+
+1. Login to the Harvester dashboard, navigate to **Advanced/Settings**
+2. Edit `csi-driver-config` by clicking `⋮`  > `Edit Setting`.
+3. Add the provisioner `rook-ceph.rbd.csi.ceph.com`.
+4. Set **Volume Snapshot Class Name** to `csi-rbdplugin-snapclass`, which refers to the name of the `VolumeSnapshotClass` used to create volume snapshots or VM snapshots.
+5. Set **Backup Volume Snapshot Class Name** to `csi-rbdplugin-snapclass`, which refers to the name of the `VolumeSnapshotClass` used to create VM backups.
```

Comment:
```suggestion
5. Set **Backup Volume Snapshot Class Name** to `csi-rbdplugin-snapclass`, which refers to the name of the `backupVolumeSnapshotClassName` used to create VM backups.
```

Reply:
Should this be `backupVolumeSnapshotClassName` instead of `VolumeSnapshotClass`?
<img width="1188" alt="Screenshot 2023-08-31 at 1 20 03 PM" src="https://github.com/harvester/harvesterhci.io/assets/83089978/44b359bf-4045-4aee-9fe8-be1743d3ff2f">

Reply:
backupVolumeSnapshotClassName is a field name.
VolumeSnapshotClass is a resource type name.
use VolumeSnapshotClass here